### PR TITLE
FIX Use DataList::orderBy()

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/10645

### CODE REVIEW

**Comment by dhensby** [comment]
Shouldn't there be a consistent interface across all lists?

**Comment by emteknetnz** [comment]
SS_List does not specify sort() (or the new orderBy() which is DataList only) - https://github.com/silverstripe/silverstripe-framework/blob/4/src/ORM/SS_List.php

We need to use DataList::orderBy() because it's specifically for using raw SQL.  DataList::sort() no longer accepts raw SQL.  That change was a security enhancement to reduce the surface area vulnerable to SQL injection attacks.

**Review by maxime-rainville — CHANGES_REQUESTED** [review]
Minimally, I would like to have a unit test to cover this scenario. 

The original error says `Uncaught InvalidArgumentException: Invalid sort column "EditableFormField"."Sort"`. Maybe, I'm being naive here, but shouldn't we be able to do something like `FooBar::get()->sort('Relation.SortField')`? Are you meant to be able to use the Dot notation in `sort`?

Side note: I'm not a big fan of how these `$sortterm`. Seems like at least part of this could made to use the regular sort.

**Comment by emteknetnz** [comment]
You can still use dot notation in sort(), in this instance it would just be `->sort('Sort')`, not `->sort('EditableFormField.Sort'`, because we're in the context of an `EditableFormField`.  However in this method there's a bunch of raw SQL being written e.g. `$sortterm .= "$col $dir, ";`, `$sortterm .= ($sortterm ? ', ' : '') . $defaultSort;`, so we need to use `orderBy()` here to retain the current logic.

Have added unit test

File: tests/GridFieldOrderableRowsTest.php [hunk 1]
```diff
@@ -346,6 +330,63 @@ private function getTitleSortedListForManipuatedData(string $dataClass, array $d
         $config->addComponent($orderable);
         $grid = new GridField('MyName', 'MyTitle', $list, $config);
         $sortedList = $orderable->getManipulatedData($grid, $list);
-        return $sortedList;
+        $col = $dataClass === TitleObject::class ? 'Title' : 'Iden';
+        $this->assertSame($expected, $sortedList->column($col));
+    }
+
+    public function provideGetManipulatedData(): array
```
Comment (maxime-rainville) [inline tests/GridFieldOrderableRowsTest.php:337]: Love the DataProvider. That's much nicer.

**Review by maxime-rainville — APPROVED** [review]
Can live with this. Tested that the new unit test fail without the fix. Feel free to Merge on Green.

> However in this method there's a bunch of raw SQL being written e.g. $sortterm .= "$col $dir, ";, $sortterm .= ($sortterm ? ', ' : '') . $defaultSort;

That's what I mean by this could easily be rewritten to work with sort. Maybe something like:
```php
$sortterm[$col] = $dir;
```

**Comment by emteknetnz** [comment]
Note the `4 of 6 checks passed` was github actions being weird, the 'context' and 'generate matrix' had duplicate jobs created which were stuck in yellow.  All the actual jobs were green.

**Pull request merged**